spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -207,6 +223,15 @@ public CustomTaskMetric[] reportDriverMetrics() {
     driverMetrics.add(TaskSkippedDataManifests.from(scanReport));
     driverMetrics.add(TaskScannedDataManifests.from(scanReport));
 
+    driverMetrics.add(TaskEqualityDeleteFiles.from(scanReport));
```

Comment:
It is a bit hard to navigate this. Can we give this block some structure as below? Let's also have some defined ordering within the group (result -> scanned -> skipped).

```
List<CustomTaskMetric> driverMetrics = Lists.newArrayList();

// common
driverMetrics.add(TaskTotalFileSize.from(scanReport));
driverMetrics.add(TaskTotalPlanningDuration.from(scanReport));

// data manifests
driverMetrics.add(TaskScannedDataManifests.from(scanReport));
driverMetrics.add(TaskSkippedDataManifests.from(scanReport));

// data files
driverMetrics.add(...);
driverMetrics.add(...);
driverMetrics.add(...);

// delete manifests
driverMetrics.add(...);
driverMetrics.add(...);

// delete files
driverMetrics.add(...);
driverMetrics.add(...);
driverMetrics.add(...);

return driverMetrics.toArray(new CustomTaskMetric[0]);
```